sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -541,4 +542,10 @@ class DateTimeUtilsSuite extends SparkFunSuite {
       }
     }
   }
+
+  test("toMillis") {
+    val input = -9223372036844776001L
+    val expected = -9223372036844777L
+    assert(DateTimeUtils.toMillis(input) == expected, s"input = ${input}")
```

Comment:
You don't need to provide a message; if you use `===` it will print a useful message if it fails. How about adding the test from the comment in line 185 above?

If you want to finish up the TODO this is looking good.

Reply:
I replaced `==` by `===`